rmgpy/rmg/main.py

```diff
@@ -1121,12 +1121,27 @@ def run_uncertainty_analysis(self):
                                     time_criteria = ([criteria.time.value], criteria.time.units)
                                     break
                             else:
-                                time_criteria = self.uncertainty['time']
+                                if self.uncertainty['time']:
+                                    time_criteria = ([self.uncertainty['time'].value], self.uncertainty['time'].units)
+                                else:
+                                    raise InputError('If terminationTime was not specified in the reactor options block, it'
+                                                    'must be specified in the uncertainty options block for global uncertainty'
+                                                    'analysis.')
+
+                                    
                             Tlist = ([reaction_system.sens_conditions['T']], 'K')
-                            Plist = ([reaction_system.sens_conditions['P']], 'Pa')
+                            try:
+                                Plist = ([reaction_system.sens_conditions['P']], 'Pa')
+                            except KeyError:
+                                #LiquidReactor
+                                Plist = ([1e8], 'Pa')
+
                             mol_frac_list = [reaction_system.sens_conditions.copy()]
                             del mol_frac_list[0]['T']
-                            del mol_frac_list[0]['P']
+                            try:
+                                del mol_frac_list[0]['P']
+                            except KeyError:
+                                pass
```

Comment:
A minor Question for Mark @amarkpayne: What is the common way to deal with things like this in RMG? Use `if` to check if a key in the `dict`, or use `try...except` to catch the key error when a key is not in a `dict`. Thanks.

Reply:
I think this is a very interesting discussion, and I would love to know @amarkpayne's thought too. 

I chose to use try/except for two reasons (correct me if I'm wrong)
(1) It is easier to ask for forgiveness than permission. (https://docs.python.org/3.4/glossary.html)
(2) For a performance point of view, `try dict[key]` will be faster than `if key in dict.keys()` **if we have lots of keys** in the dictionary because `dict[key]` is a hash table so it's O(1) and `if key in dict.keys()` is a list so it's O(n). (Of course we won't have lots of keys in this dict so this will barely make a difference)

If we have some convention in RMG, I will happily change it!

Reply:
There's no defined way to do this in RMG. There some discussion of the pluses and minuses here:  https://stackoverflow.com/questions/4512557/if-key-in-dict-vs-try-except-which-is-more-readable-idiom

My personal opinion is unless we think it might be important for performance we should just use the conditional statement. But it doesn't matter than much.